src/System.Management.Automation/engine/Attributes.cs

```diff
@@ -1114,6 +1114,11 @@ public sealed class ValidatePatternAttribute : ValidateEnumeratedArgumentsAttrib
         public RegexOptions Options { set; get; } = RegexOptions.IgnoreCase;
 
         /// <summary>
+        /// Gets or sets the custom error message that is displayed to the user
+        /// </summary>
+        public string ErrorMessage { get; set; }
```

Comment:
I do see one argument against - we can't require the derived classes all use the `ErrorMessage` property when they format their messages. So it would be misleading to say the property is supported on any validation attribute.